crypto/evp/evp_pkey.c

```diff
@@ -73,6 +73,15 @@ EVP_PKEY *EVP_PKCS82PKEY_ex(const PKCS8_PRIV_KEY_INFO *p8, OSSL_LIB_CTX *libctx,
     int selection;
     size_t len;
     OSSL_DECODER_CTX *dctx = NULL;
+    const ASN1_OBJECT *algoid = NULL;
+    const char *keytype = NULL;
+
+    if (PKCS8_pkey_get0(&algoid, NULL, NULL, NULL, p8))
+        keytype = OBJ_nid2sn(OBJ_obj2nid(algoid));
+    /*
+     * else we don't know the OID...carry on anyway to see if we may have a
+     * decoder that can handle it anyway
+     */
```

Comment:
I would rather see that we use `OBJ_obj2txt`

Reply:
Why do you see that as better? The advantage of this construction is that it gives you a `const char * `to the name directly. `OBJ_obj2txt` introduces an additional buffer - which seems over the top for this.

Reply:
Consider the case where a provider provides a key type that we know nothing about, but have included the OID in the OSSL_ALGORITHM name string.  That makes it directly fetchable, even if our OBJ db doesn't have a record for it.